stardao-mongodb/src/main/java/io/stardog/stardao/mongodb/AbstractMongoDao.java

```diff
@@ -93,6 +93,22 @@ protected Object generateId() {
     }
 
     /**
+     * Load a partial object that might not exist by id.
+     * @param id    primary key value
+     * @param fields    set of fields to return
+     * @return
+     */
+    public Optional<P> loadOpt(K id, Iterable<String> fields) {
```

Comment:
Note that this method is not simply an overload - it returns an `Optional<P>` (Partial) instead of an `Optional<M>` (Model). Because we can't guarantee that the user passed in all the required fields so we have to return a Partial.

I see the purpose of applying projection on every `load()`. It could help in some situations where you have old fields hanging about that you no longer care about and are no longer in the model. But in a lot of situations it would be identical to omitting the projection (since all the fields would be present). Are we absolutely sure that including a projection for all fields doesn't cause MongoDB some kind of performance overhead? Might be worth doing a test before we make this the default.

I also wonder if we should cache the result of `storageFieldsProjection()` since it seems slightly wasteful to rebuild that map every time.

Also your code made me realize I screwed up the fields -- the public interface should use the model field names and translate it to the storageNames. Will fix that for both DynamoDB and MongoDB.